api_app/analyzers_manager/observable_analyzers/dns/dns_malicious_detectors/spamhaus_wqs.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+from ..dns_responses import malicious_detector_response
+
+logger = logging.getLogger(__name__)
+
+
+class SpamhausWQS(classes.ObservableAnalyzer):
+    url: str = "https://apibl.spamhaus.net/lookup/v1"
+    _api_key: str = None
+
+    def update(self):
+        pass
+
+    def run(self):
+        headers = {"Authorization": f"Bearer {self._api_key}"}
+
+        if self.observable_classification == self.ObservableTypes.DOMAIN.value:
+            response = requests.get(
+                url=f"{self.url}/DBL/{self.observable_name}", headers=headers
+            )
+        elif self.observable_classification == self.ObservableTypes.IP.value:
+            response = requests.get(
+                url=f"{self.url}/AUTHBL/{self.observable_name}", headers=headers
+            )
+
+        if response.status_code == 200:
+            # 200 - Found - The record is listed
+            return malicious_detector_response(self.observable_name, True)
+        elif response.status_code == 404:
+            # 404 - Not found - The record is not listed
+            return malicious_detector_response(self.observable_name, False)
+        else:
+            raise AnalyzerRunException(f"result not expected: {response}")
```

Comment:
you print the response object here. Instead you should print the status code

Reply:
also, how much are you sure that these are the only status codes allowed?

Reply:
[here](https://docs.spamhaus.com/datasets/docs/source/70-access-methods/web-query-service/060-api-info.html#http-response-status-codes) is a list of status codes allowed.

Reply:
can you link this in the code as a comment? so we can keep in mind the reason in the future.

Reply:
updated